mods/sethome/init.lua

```diff
@@ -24,42 +25,58 @@ loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
+function sethome.go_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		-- just a check to prevent the server crashing
+		return false
+	end
+	if homepos[name] then
+		player:setpos(homepos[name])
+		minetest.chat_send_player(name, "Teleported to home!")
+		return true
+	else
+		minetest.chat_send_player(name, "Set a home using /sethome")
+		return false
+	end
+end
 
 minetest.register_chatcommand("home", {
     description = "Teleport you to your home point",
     privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+    func = sethome.go_home,
 })
 
+function sethome.set_home(name, p)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		return false
+	end
+	local pos = p
+	if not p or p == "" then
+		pos = player:getpos()
+	end
```

Comment:
p should never be "". Also, you can set the parameter without it effecting the caller, so all that is needed is:

```lua
function sethome.set_home(name, pos)
	-- stuff
	if not pos then
		pos = player:getpos()
	end
```